keras_cv/bounding_box/mask_invalid_detections.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+from keras_cv.bounding_box.to_dense import to_dense
+from keras_cv.bounding_box.to_ragged import to_ragged
+from keras_cv.bounding_box.validate_format import validate_format
+
+
+def mask_invalid_detections(bounding_boxes):
+    """masks out invalid detections with -1s.
+
+    This utility is mainly used on the output of `tf.image.combined_non_max_suppression`
+    operations.  The output of `tf.image.combined_non_max_suppression` contains padding
+    values of 0 instead of -1.  KerasCV expects all bounding boxes to be padded with
+    -1s instead of 0s.  This function uses the value of `num_detections` to mask out
+    invalid boxes with -1s.
+
+    Args:
+        bounding_boxes: a dictionary complying with KerasCV bounding box format.  In
+            addition to the normal required keys, these boxes are also expected to have
+            a `num_detections` key.
+
+    Returns:
+        bounding boxes with proper masking of the boxes according to `num_detections`.
+        This allows proper interop with `tf.image.combined_non_max_suppression`.
+        Returned boxes match the specification fed to the function, so if the bounding
+        box tensor uses `tf.RaggedTensor` to represent boxes the returned value will
+        also return `tf.RaggedTensor` representations.
+    """
+    # ensure we are complying with KerasCV bounding box format.
+    info = validate_format(bounding_boxes)
+    if "num_detections" not in bounding_boxes:
+        raise ValueError(
+            "`bounding_boxes` must have key 'num_detections' "
+            "to be used with `bounding_box.mask_invalid_detections()`."
+        )
+
+    # required to use tf.where()
+    bounding_boxes = to_dense(bounding_boxes)
```

Comment:
Shouldn't we expect the boxes to be dense as a precondition, since if they were ragged we wouldn't need to mask them this way?

Reply:
sure, updated